nimble/host/src/ble_eatt.c

```diff
@@ -355,7 +441,22 @@ ble_gatt_eatt_write_cl_cb(uint16_t conn_handle,
         return 0;
     }
 
-    ble_eatt_start(conn_handle);
+#if (MYNEWT_VAL(BLE_EATT_AUTO_CONNECT))
+    struct ble_gap_conn_desc desc;
+    int rc;
+
+    rc = ble_gap_conn_find(conn_handle, &desc);
+    assert(rc == 0);
+
+    if (desc.role == BLE_GAP_ROLE_SLAVE) {
+        /* Add initial delay as peripheral to avoid collision */
+        conn_cb_arg.conn_handle = conn_handle;
+        os_callout_reset(&eatt_conn_timer, 500 * OS_TICKS_PER_SEC / 1000);
```

Comment:
I'd suggest to use same value as spec recommends for  GATT collisions mitigation resolving

"Peripheral shall  wait a minimum of 100 ms before retrying; on LE connections, the Peripheral shall wait
at least 2 × (connPeripheralLatency + 1) × connInterval if that is longer.  "